### ISSUE TITLE
ansible-lint[risky-shell-pipe] misinterprets pipe in comment when using Multiline Block Scalars

### ISSUE THREAD

**Issue by jedthe3rd** [ansible/ansible-lint#4262]
##### Summary

The ansible-lint[risky-shell-pipe] rule misinterprets pipe in comment when using Multiline Block Scalars

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
> ansible-lint --version
ansible-lint 24.7.0 using ansible-core:2.17.1 ansible-compat:24.7.0 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

- ansible installation method: ansible-dev-tools devcontainer - ghcr.io/ansible/community-ansible-dev-tools:latest
- ansible-lint installation method: ansible-dev-tools devcontainer - ghcr.io/ansible/community-ansible-dev-tools:latest

##### STEPS TO REPRODUCE

Create the below test.yml and run it through the linter
```Yaml
---
- name: "Write crontab header for {{ postgresql_svc_user }}"
  become: true
  ansible.builtin.shell: |
    cat <<EOF >> crontab_temp.txt ##    |    +---------------- hour (0 - 23)
  changed_when: true
...

```

##### Desired Behavior

Expected behavior is the pipes in comments are interpreted correctly and not flagged for risky-shell-pipe rule.

##### Actual Behavior

![image](https://github.com/user-attachments/assets/771f7eaf-7a26-40e4-9bfe-ddaa10d4b382)

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by cidrblock** [comment]
+100 for using the devtools container, ty for that!
